File: generator/src/test/resources/kotlin/expected/sunday/RequestUriParamsTest/test-generation-of-multiple-uri-parameters-with-inline-type-definitions.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.Operation
import io.outfoxx.sunday.OperationSpec
import io.outfoxx.sunday.Transport
import io.outfoxx.sunday.http.Method
import io.outfoxx.sunday.http.Request
import io.outfoxx.sunday.operation
import kotlin.Any
import kotlin.String
import kotlin.Unit
import kotlin.collections.List
import kotlin.collections.Map

public class API<Req : Request>(
  public val transport: Transport<Req>,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public fun fetchTest(category: FetchTestCategoryUriParam, type: FetchTestTypeUriParam):
      Operation<Unit, Map<String, Any>, Req> =
      this.transport.operation<Unit, Map<String, Any>, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/tests/{category}/{type}",
      pathParameters = mapOf(
        "category" to category,
        "type" to type
      ),
      acceptTypes = this.defaultAcceptTypes
    )
  )

  public enum class FetchTestCategoryUriParam(
    private val wireValue: String,
  ) {
    Politics("politics"),
    Science("science"),
    ;

    public override fun toString(): String = wireValue
  }

  public enum class FetchTestTypeUriParam(
    private val wireValue: String,
  ) {
    All("all"),
    Limited("limited"),
    ;

    public override fun toString(): String = wireValue
  }
}

File: generator/src/test/resources/swift/expected/RequestUriParamsTest/test-generation-of-multiple-uri-parameters-with-inline-type-definitions.output.swift
import PotentCodables
import Sunday

public final class API<TransportType : Transport> : Sendable {

  public static var problemTypes: [ProblemRegistration] {
    return []
  }
  public let transport: TransportType
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    transport: TransportType,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json],
    problemTypes: [ProblemRegistration] = API.problemTypes
  ) {
    self.transport = transport
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
    problemTypes.forEach { $0.register(on: transport) }
  }

  public func fetchTest(category: FetchTestCategoryUriParam, type: FetchTestTypeUriParam) throws -> Sunday.Operation<Empty, [String : AnyValue], TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/tests/{category}/{type}",
        pathParameters: [
          "category": try ParameterValues.encode(category),
          "type": try ParameterValues.encode(type)
        ],
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

  public enum FetchTestCategoryUriParam : String, CaseIterable, Codable, Sendable {

    case politics = "politics"
    case science = "science"

  }

  public enum FetchTestTypeUriParam : String, CaseIterable, Codable, Sendable {

    case all = "all"
    case limited = "limited"

  }

}
